server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -876,61 +853,53 @@ private void cancelOfflineTableMigrations(KeyExtent extent) {
+        scanner.setRange(new Range(extent.toMetaRow()));
+        entries = scanner.stream();
       }
-      Scanner scanner = manager.getContext().createScanner(table, Authorizations.EMPTY);
-      scanner.fetchColumnFamily(CurrentLocationColumnFamily.NAME);
-      scanner.fetchColumnFamily(FutureLocationColumnFamily.NAME);
-      scanner.setRange(new Range(row));
-      for (Entry<Key,Value> entry : scanner) {
+
+      entries.forEach(entry -> {
         if (entry.getKey().getColumnFamily().equals(CurrentLocationColumnFamily.NAME)) {
           assigned.put(entry.getKey(), entry.getValue());
         } else if (entry.getKey().getColumnFamily().equals(FutureLocationColumnFamily.NAME)) {
           future.put(entry.getKey(), entry.getValue());
         }
-      }
-      if (!future.isEmpty() && !assigned.isEmpty()) {
-        Manager.log.warn("Found a tablet assigned and hosted, attempting to repair");
-      } else if (future.size() > 1 && assigned.isEmpty()) {
-        Manager.log.warn("Found a tablet assigned to multiple servers, attempting to repair");
-      } else if (future.isEmpty() && assigned.size() > 1) {
-        Manager.log.warn("Found a tablet hosted on multiple servers, attempting to repair");
+      });
+
+      var count = future.size() + assigned.size();
+      if (count <= 1) {
+        Manager.log.trace("Tablet {} seems to have correct location based on inspection",
+            tabletMetadata.getExtent());
       } else {
-        Manager.log.info("Attempted a repair, but nothing seems to be obviously wrong. {} {}",
-            assigned, future);
-        return;
-      }
-      Iterator<Entry<Key,Value>> iter =
-          Iterators.concat(future.entrySet().iterator(), assigned.entrySet().iterator());
-      while (iter.hasNext()) {
-        Entry<Key,Value> entry = iter.next();
-        TServerInstance alive = manager.tserverSet.find(entry.getValue().toString());
-        if (alive == null) {
-          Manager.log.info("Removing entry  {}", entry);
-          BatchWriter bw = manager.getContext().createBatchWriter(table);
-          Mutation m = new Mutation(entry.getKey().getRow());
-          m.putDelete(entry.getKey().getColumnFamily(), entry.getKey().getColumnQualifier());
-          bw.addMutation(m);
-          bw.close();
-          return;
+        for (Map.Entry<Key,Value> entry : future.entrySet()) {
+          TServerInstance alive = manager.tserverSet.find(entry.getValue().toString());
+          Manager.log.debug("Saw duplicate future location key:{} value:{} alive:{} ",
+              entry.getKey(), entry.getValue(), alive != null);
+        }
+        for (Map.Entry<Key,Value> entry : assigned.entrySet()) {
+          TServerInstance alive = manager.tserverSet.find(entry.getValue().toString());
+          Manager.log.debug("Saw duplicate current location key:{} value:{} alive:{} ",
+              entry.getKey(), entry.getValue(), alive != null);
         }
```

Comment:
```java
  private void logDuplicateLocations(Map<Key,Value> locations, String type) {
    for (Map.Entry<Key,Value> entry : locations.entrySet()) {
      TServerInstance alive = manager.tserverSet.find(entry.getValue().toString());
      Manager.log.debug("Saw duplicate {} location key:{} value:{} alive:{}", type, entry.getKey(),
          entry.getValue(), alive != null);
    }
  }
```
Could refactor these two loops into one method

Reply:
Looked into doing this and realized the column family in the key will indicate the type, so the two maps were redundant.  Collapsed it to a single map [e0e2e92](https://github.com/apache/accumulo/pull/4668/commits/e0e2e929f95ab95d1f4dd2659dc2822edb9bcd7d)